src/herbivoresAndCarnivores.js

```diff
@@ -2,14 +2,35 @@
 
 class Animal {
   // write your code here
+  static alive = [];
+  health = 100;
+  constructor(name) {
+    this.name = name;
+    Animal.alive.push(this);
+  }
 }
 
 class Herbivore extends Animal {
   // write your code here
+  hidden = false;
+  hide() {
+    this.hidden = true;
+  }
 }
 
 class Carnivore extends Animal {
   // write your code here
+  bite(target) {
+    if (!target.hidden) {
+      if (!(target instanceof Carnivore)) {
```

Comment:
The condition `!(target instanceof Carnivore)` should be `target instanceof Herbivore` to ensure that only herbivores can be bitten by carnivores, as per typical predator-prey relationships.